packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -376,6 +388,77 @@ contract L2Genesis is Script, Artifacts {
+        // so should genesisTime = cfg.l2BlockTime?
+        // block.timestamp = 1
+        uint256 genesisTime = block.timestamp;
+
+        string memory serializedConfigName = "chainConfig";
+        string memory json = "";
+        json = stdJson.serialize(serializedConfigName, "chainId", cfg.l2ChainID());
+        json = stdJson.serialize(serializedConfigName, "homesteadBlock", uint256(0));
+        // TODO DAOForkSupport is probably not supposed to be `false`, if `false` in
+        // deploy config, should we just not include it?
+        json = stdJson.serialize(serializedConfigName, "DAOForkSupport", false);
+        json = stdJson.serialize(serializedConfigName, "eip150Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "eip155Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "eip158Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "byzantiumBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "constantinopleBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "petersburgBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "istanbulBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "muirGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "berlinBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "londonBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "arrowGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "grayGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "mergeNetsplitBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "terminalTotalDifficulty", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "terminalTotalDifficultyPassed", true);
+        // TODO This was previously set as uint64, is it okay to be uint256?
+        json = stdJson.serialize(serializedConfigName, "bedrockBlock", cfg.l2GenesisBlockNumber());
+        json = stdJson.serialize(serializedConfigName, "regolithTime", _getHardforkTime(cfg.l2GenesisRegolithTimeOffset(), genesisTime));
+        json = stdJson.serialize(serializedConfigName, "canyonTime", _getHardforkTime(cfg.l2GenesisCanyonTimeOffset(), genesisTime));
+
+        // TODO
+        json = stdJson.serialize(serializedConfigName, "shanghaiTime", uint256(0));
+        // TODO
+        json = stdJson.serialize(serializedConfigName, "cancunTime", uint256(0));
+        // TODO
+        json = stdJson.serialize(serializedConfigName, "ecotoneTime", uint256(0));
+        // TODO
+        json = stdJson.serialize(serializedConfigName, "interopTime", uint256(0));
+
+        string memory serializedOptimismConfig = "optimismConfig";
+        string memory nestedOptimismConfig = "";
+        // TODO
```

Comment:
So really what we want here are defaults of these values with overrides. That is kind of clunky to express in solidity. Perhaps a function with a try catch or if else in the deploy config instead of reading in the `read` function and assigning to a public variable

Reply:
I've added [these methods](https://github.com/ethereum-optimism/optimism/pull/9328/files#diff-fcedbe6005fe092aab124a65c9eb9a2c5824c1cd83461b144daea68b6c8a4d2bR298-R319) to do as you've suggested - attempt to read the properties, supply a default if not specified in deploy config